scripts/lambda/daily_analytics.py

```diff
@@ -0,0 +1,141 @@
+def calculate_command_activity(last_to_date):
+    commands_to_date = {}
+    last_commands_to_date = last_to_date.get("command_activity", {})
+    for doc in db.analytics_command_activity.find():
+        commands_to_date[doc['name']] = doc['num_invocations']
+
+    commands_today = {}
+    for command, to_date in commands_to_date.items():
+        commands_today[command] = to_date - last_commands_to_date.get(command, 0)
+
+    return commands_today, commands_to_date
+
+
+# each of these should return (total_today, total_life) [int, int]
+def calculate_num_commands(last_to_date):
+    num_commands_now = get_statistic("commands_used_life")
+    delta = num_commands_now - last_to_date.get("num_commands", 0)
+    return delta, num_commands_now
+
+
+def calculate_num_characters(last_to_date):
+    num_characters_now = db.characters.estimated_document_count()
+    delta = num_characters_now - last_to_date.get("num_characters", 0)
+    return delta, num_characters_now
+
+
+# returns {day: int, week: int, month: int}
+def calculate_num_active_users():
+    now = datetime.datetime.now()
+    out = {
+        "day": db.analytics_user_activity.count_documents({"last_command_time":
+                                                               {"$gt": now - datetime.timedelta(days=1)}}),
+        "week": db.analytics_user_activity.count_documents({"last_command_time":
+                                                                {"$gt": now - datetime.timedelta(days=7)}}),
+        "month": db.analytics_user_activity.count_documents({"last_command_time":
+                                                                 {"$gt": now - datetime.timedelta(days=30)}})
+    }
+    return out
+
+
+def calculate_num_active_guilds():
+    now = datetime.datetime.now()
+    out = {
+        "day": db.analytics_guild_activity.count_documents({"last_command_time":
+                                                                {"$gt": now - datetime.timedelta(days=1)}}),
+        "week": db.analytics_guild_activity.count_documents({"last_command_time":
+                                                                 {"$gt": now - datetime.timedelta(days=7)}}),
+        "month": db.analytics_guild_activity.count_documents({"last_command_time":
+                                                                  {"$gt": now - datetime.timedelta(days=30)}})
+    }
+    return out
+
+
+# main
+def calculate_daily():
+    try:
+        last = next(db.analytics_daily.find().sort("timestamp", pymongo.DESCENDING).limit(1))
+    except StopIteration:
+        last = {}
+    last_to_date = last.get("to_date", {})
+
+    # setup
+    out = {"timestamp": datetime.datetime.now()}
+    to_date = dict()
+
+    # --- calculations ---
+
+    # -- deltas --
+    # most popular commands today
+    # fixme: how does this work in a columnal data store?
+    # out['command_activity'], to_date['command_activity'] = calculate_command_activity(last_to_date)
```

Comment:
The code to calculate these is above (`calculate_command_activity()`), but it outputs a dict that looks like:
```
{
    "roll": 10321,
    "check": 7130,
    ... (~70 entries)
}
```
that just doesn't work well for a columnal data store.